src/pyodide/internal/python.js

```diff
@@ -199,10 +206,10 @@ async function instantiateEmscriptenModule(emscriptenSettings) {
     // removed =(
     // If/when we link our own Pyodide we can remove this.
     globalThis.window = {}; // makes ENVIRONMENT_IS_WEB    = true
-    globalThis.importScripts = 1; // makes ENVIRONMENT_IS_WORKER = false
-    const p = _createPyodideModule(emscriptenSettings);
+    globalThis.importScripts = 1;
+    globalThis.navigator = { userAgent: "Safari" };
+    const p = createPyodideModule(emscriptenSettings);
```

Comment:
Can you check if it works to delete these after the call to `createPyodideModule`?